lib/galaxy/tool_util/deps/container_classes.py

```diff
@@ -178,7 +178,7 @@ def add_var(name, value):
             if self.job_info.tool_directory:
                 defaults += ",$tool_directory:default_ro"
             if self.job_info.job_directory:
-                defaults += ",$job_directory:default_ro"
+                defaults += ",$job_directory:default_ro,$job_directory/outputs:rw"
```

Comment:
So do singularity and docker handle that correctly ? I think there's a change they may not :/

Reply:
The working directory uses the same trick though right? Everything is broken if this doesn't work 😿.

Reply:
I have to admit that I cheated when I deployed singularity and just mounted everything rw, but I can check

Reply:
I think Singularity is a known quantity and we work around it?

> Galaxy allows the mount try "default_ro" which translates to ro for Docker and ro for Singularity iff no subdirectories are rw (Singularity does not allow ro parent directories with rw subdirectories).

Reply:
ok, I think that's fine then.